crates/fuel-core/src/database.rs

```diff
@@ -421,11 +239,8 @@ impl AtomicView for Database<OnChain> {
 
     type Height = BlockHeight;
 
-    fn latest_height(&self) -> BlockHeight {
-        // TODO: The database should track the latest height inside of the database object
-        //  instead of fetching it from the `FuelBlocks` table. As a temporary solution,
-        //  fetch it from the table for now.
-        self.latest_height().unwrap_or_default()
+    fn latest_height(&self) -> Option<Self::Height> {
+        *self.height.lock().expect("poisoned")
```

Comment:
For these `expect`s, Instead of telling why it panicked, we should explain why this shouldn't panic. I'm skeptical.